pkg/apis/options/legacy_options.go

```diff
@@ -119,3 +129,250 @@ func (l *LegacyUpstreams) convert() (Upstreams, error) {
 
 	return upstreams, nil
 }
+
+type LegacyHeaders struct {
+	PassBasicAuth     bool `flag:"pass-basic-auth" cfg:"pass_basic_auth"`
+	PassAccessToken   bool `flag:"pass-access-token" cfg:"pass_access_token"`
+	PassUserHeaders   bool `flag:"pass-user-headers" cfg:"pass_user_headers"`
+	PassAuthorization bool `flag:"pass-authorization-header" cfg:"pass_authorization_header"`
+
+	SetBasicAuth     bool `flag:"set-basic-auth" cfg:"set_basic_auth"`
+	SetXAuthRequest  bool `flag:"set-xauthrequest" cfg:"set_xauthrequest"`
+	SetAuthorization bool `flag:"set-authorization-header" cfg:"set_authorization_header"`
+
+	PreferEmailToUser    bool   `flag:"prefer-email-to-user" cfg:"prefer_email_to_user"`
+	BasicAuthPassword    string `flag:"basic-auth-password" cfg:"basic_auth_password"`
+	SkipAuthStripHeaders bool   `flag:"skip-auth-strip-headers" cfg:"skip_auth_strip_headers"`
```

Comment:
It looks like we lost this code path in the refactor. I don't think I see any of this security logic.

I also had a TODO to flip this to `true` by default in this breaking release here: https://github.com/oauth2-proxy/oauth2-proxy/pull/679

It looks l had taken a try at cleaning up a couple other of these flag's inconsistencies. Let me know if you want to take that one here, or I can make a follow up PR to this once this merges.

Reply:
Here's the missing logic: https://github.com/oauth2-proxy/oauth2-proxy/pull/826/files#diff-7e1e132991c10dc0110f813e8e1057f24288bfb9d7d5bdd54d468c496e01110bL1081

Reply:
I've just updated, I don't think the strip auth is missing necessarily, I had forgotten to convert the flag which is now fixed.

If you go through any path that should add headers (or even the SkipAuthProxy) you go through the header chain, which should strip the headers based on the `PreserveRequestValue` field which defaults to false, but is converted as true to maintain existing behaviour